docs/get-started/clinc150/index.md

```diff
@@ -0,0 +1,842 @@
+But how about intent accuracy? How many times do we predict the correct intent? What about
+predicting intents in the first place?
+
+In an intent classification task, we want to classify utterances to intents.
+So far we formulated the task as a search problem, but to actually use the model
+we need to revert back to the classification task. The missing part is a
+decision function that can produce intent classes on a test utterance, by
+utilising the finetuned embedding model.
+
+A straight-forward way to do that, is to embed the test utterance using our
+fine-tuned model and search for the nearest neighbor from a set of utterances
+with pre-computed embeddings. This set of utterances with pre-computed
+embeddings is usually refered to as the index, and we can use our training
+data for it. The class that the nearest neighbor belongs to, will be the class
+that we assign to the test utterance.
+
+To go one step further, we can also choose to fetch multiple neighbors
+to the test utterance and decide on the intents to return, using a simple rule
+that takes into account both the class and the distance of each neighbor.
+
+We implement this function, using `docarray`s `match` and `finetuner`s `embed`
+methods.
+
+```python
+from collections import defaultdict
+from typing import Tuple
+
+from finetuner import embed
+
+
+def predict_intents(
+    utterance: str,
+    model: torch.nn.Module,
+    index: DocumentArray,
+    k: int = 20,
+) -> List[Tuple[str, float]]:
+    """
+    Find top k nearest neighbors in a search query
+    and compute intents by aggregating distances
+    """
+    doc = Document(text=utterance)
+    embed(
+        DocumentArray(doc),
+        embed_model=model,
+        device=DEVICE,
+        batch_size=1,
+        collate_fn=collate_fn,
+    )
+    doc.match(index, limit=k)
+
+    intents = [m.tags['finetuner_label'] for m in doc.matches]
+    distances = [m.scores['cosine'].value for m in doc.matches]
+    sum_distances = sum(distances)
+    scores = [dist / sum_distances for dist in distances]
+
+    output = defaultdict(float)
+    for intent, score in zip(intents, scores):
+        output[intent] += score
+
+    return sorted(list(output.items()), key=lambda x: x[1], reverse=True)
+```
+
+Let's index our training data:
+
+```python
+from copy import deepcopy
+
+pretrained_model = TransformerEmbedder()
+
+# First let's index our data
+pretrained_index = deepcopy(train_data)
+finetuned_index = deepcopy(train_data)
+
+embed(
+    pretrained_index,
+    embed_model=pretrained_model,
+    device=DEVICE,
+    batch_size=BATCH_SIZE,
+    collate_fn=collate_fn,
+)
+embed(
+    finetuned_index,
+    embed_model=finetuned_model,
+    device=DEVICE,
+    batch_size=BATCH_SIZE,
+    collate_fn=collate_fn,
+)
+```
+
+Let's now use this method to run some test cases.
+
+```python
+utterance = 'Where do you thing I should travel to this Christmas?'
```

Comment:
Is it intentional? 
```suggestion
utterance = 'Where do you think I should travel to this Christmas?'
```

Reply:
Nope 😛  Now I need to re-run the tutorial